### ISSUE TITLE
sb-rn-get-stories does not support stories config with a configuration object (titlePrefix)

### ISSUE THREAD

**Issue by stevoland** [storybookjs/react-native#423]
**Describe the bug**

`6.0.1-beta.12` supports auto titles (thanks!) but using `titlePrefix` requires using [a configuration object](https://storybook.js.org/docs/react/configure/overview#with-a-configuration-object) which `sb-rn-get-stories` does not handle.

Happy to contribute if you can confirm I'm on the right path:

Assume I could use [this API](https://github.com/storybookjs/storybook/blob/next/code/lib/preview-api/src/modules/store/autoTitle.ts#L48) in `scripts/loader.js` to handle configuration object configs and generate a `storybook.requires.ts` like:

```js
// ...
const getStories = () => {
  return {
    'Prefix/Component': require('./stories/Component.stories.tsx'),
  }
}
// ...
```

**To Reproduce**
```js
// .storybook/main.js
module.exports = {
  stories: [
    {
      directory: './stories',
      files: '**/*.stories.tsx',
      titlePrefix: 'Prefix',
    },
  ]
}
```

```
$ yarn sb-rn-get-stories
...
/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/minimatch/minimatch.js:275
    throw new TypeError('invalid pattern')
    ^

TypeError: invalid pattern
    at assertValidPattern (/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/minimatch/minimatch.js:275:11)
    at new Minimatch (/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/minimatch/minimatch.js:133:3)
    at setopts (/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/glob/common.js:116:20)
    at new GlobSync (/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/glob/sync.js:37:3)
    at Function.globSync [as sync] (/Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/node_modules/glob/sync.js:23:10)
    at /Users/stephen.collings/workspace/native/node_modules/@storybook/react-native/scripts/loader.js:85:24
...
```

**Expected behavior**
Generates a navigation structure like:

Prefix > Component

**System:**
```
System:
    OS: macOS 12.5.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.18.1/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.1/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 108.0.1
    Safari: 15.6.1
  npmPackages:
    @storybook/addon-actions: 6.5.14 => 6.5.14
    @storybook/addon-essentials: 6.5.14 => 6.5.14
    @storybook/addon-knobs: 6.4.0 => 6.4.0
    @storybook/addon-links: 6.5.14 => 6.5.14
    @storybook/addon-ondevice-actions: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/addon-ondevice-backgrounds: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/addon-ondevice-controls: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/addon-ondevice-notes: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/addons: 6.5.14 => 6.5.14
    @storybook/react: 6.5.14 => 6.5.14
    @storybook/react-native: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/react-native-server: 6.0.1-beta.12 => 6.0.1-beta.12
    @storybook/theming: 6.5.14 => 6.5.14
```

**Comment by dannyhw** [comment]
@stevoland hey thanks for raising this and offering your help, I would really appreciate that! Yes you are correct, the changes to support title generation were recently added and they are in scripts/loader.js.

**Comment by SethY8s** [comment]
@dannyhw After upgrading to the latest release I am having similar problems even with a normal path. Down below is my main.js and I have kept it the same throughout working on storybook.

```
`module.exports = {
    stories: [
        '../src/components/**/*.stories.?(tsx)',
    ],
    addons: [
        '@storybook/addon-ondevice-notes',
        '@storybook/addon-ondevice-controls',
        '@storybook/addon-ondevice-backgrounds',
        '@storybook/addon-ondevice-actions',
    ],
};`
```
Now, whenever I run "yarn sb-rn-get-stories --config-path .ondevice"(same as your expo template) I get an error saying:

SyntaxError: Bad character escape sequence. (42:1792)
  40 |
  41 |       const getStories=() => {... all my stories

strangely enough, my co-worker on the same branch is able to run it, however, he is on macOS instead of windows. I tried all the normal fixes, such as deleting node_modules, pulling from his branch to make sure everything is configured the same etc. Do you think what @stevoland added will fix this issue or do you think there is something on my side such as a node version difference? Another important thing to note is that changing my main.js to a more direct path like this still works:

```
`module.exports = {
    stories: [
        '../src/components/comp1/**/*.stories.?(tsx)',
        '../src/components/comp2/**/*.stories.?(tsx)',
        '../src/components/comp3/**/*.stories.?(tsx)',
    ],
    addons: [
        '@storybook/addon-ondevice-notes',
        '@storybook/addon-ondevice-controls',
        '@storybook/addon-ondevice-backgrounds',
        '@storybook/addon-ondevice-actions',
    ],
};`
```
 I

**Comment by stevoland** [comment]
Resolved in [v6.5.0-rc.1](https://github.com/storybookjs/react-native/releases/tag/v6.5.0-rc.1). Thanks @dannyhw 🙏
